### PR TITLE
Visualize multiple traces

### PR DESCRIPTION
**Pull request by thowell**
Adds support for visualizing multiple traces. 
- Visualizing a single trace now requires specifying "trace0".
- Additional traces are specified with an index, e.g., "trace1", "trace2", etc.

Minor fixes:
- Turn on iLQG action limits by default (this was accidentally turned off--fortunately, doesn't seem to make a meaningful difference on any examples I tested)
- Increase `kMaxCostTerms = 30` + a warning if this value is exceeded by the xml cost specification.
- Set time in `ModelDerivatives::Compute`
- Set quadruped hill task floor to default

### CODE REVIEW

File: mjpc/task.h [hunk 1]
```diff
@@ -68,12 +68,13 @@ class Task {
 
   // cost parameters
   int num_residual;
-  int num_norms;
+  int num_cost;
+  int num_trace;
   std::vector<int> dim_norm_residual;
-  std::vector<int> num_norm_parameters;
+  std::vector<int> num_num_parameter;
```
Comment (nimrod-gileadi) [inline mjpc/task.h:74]: Can you add a comment on what this number is? Weird name.

**Review by nimrod-gileadi — DISMISSED** [review]

File: mjpc/utilities.cc [hunk 2]
```diff
@@ -83,13 +84,36 @@ double* SensorByName(const mjModel* m, const mjData* d,
...
+
+  // get first trace, either "trace" or "trace0"
+  double* trace = SensorByName(m, d, "trace");
+  if (!trace) {
+    trace = SensorByName(m, d, "trace0");
+  }
+  if (trace) mju_copy(traces, trace, 3);
+
+  // allocate string
+  char str[7];
```
[the remark this answers is not in the document]
Comment (thowell) [reply mjpc/utilities.cc:104]: Done. New constant: `kMaxTraces = 99`
Also, added a check when the task is constructed from xml.

File: mjpc/utilities.cc [hunk 3]
```diff
@@ -83,13 +84,36 @@ double* SensorByName(const mjModel* m, const mjData* d,
...
+  for (int i = 1; i < num_trace; i++) {
+    // set trace id
+    mujoco::util_mjpc::sprintf_arr(str, "trace%i", i);
+    trace = SensorByName(m, d, str);
+    if (trace) mju_copy(traces + 3 * i, trace, 3);
```
[the remark this answers is not in the document]
Comment (thowell) [reply mjpc/utilities.cc:109]: The trajectory traces already default to zeros. If the sensor call fails the trace is already filled with zeros.

File: mjpc/utilities.cc [hunk 4]
```diff
@@ -83,13 +84,36 @@ double* SensorByName(const mjModel* m, const mjData* d,
...
   }
 }
 
+// get traces from sensors
+void GetTraces(double* traces, const mjModel* m, const mjData* d, int num_trace) {
+  if (num_trace == 0) {
+    return;
+  }
+
+  // get first trace, either "trace" or "trace0"
```
[the remark this answers is not in the document]
Comment (thowell) [reply mjpc/utilities.cc]: Changed to only look for "trace0" for simplicity. Updated the docs and tasks.

File: mjpc/utilities.cc [hunk 5]
```diff
@@ -83,13 +84,36 @@ double* SensorByName(const mjModel* m, const mjData* d,
                      const std::string& name) {
   int id = mj_name2id(m, mjOBJ_SENSOR, name.c_str());
   if (id == -1) {
-    std::cerr << "sensor \"" << name << "\" not found.\n";
+    // std::cerr << "sensor \"" << name << "\" not found.\n";
```
[the remark this answers is not in the document]
Comment (thowell) [reply mjpc/utilities.cc]: Fixed. 
(No longer an issues since we only look for "trace0" now.)

File: mjpc/task.h (hunk 1 above)
Comment (thowell) [reply mjpc/task.h]: Bad find-and-replace... Fixed to `num_norm_parameter`.

**Comment by thowell** [comment]
@nimrod-gileadi I addressed your comments and updated the original post to clarify the final changes.

**Review by nimrod-gileadi — APPROVED** [review]

**Pull request merged**